### ISSUE TITLE
Each deployment takes about 12 hours on Azure App Service

### ISSUE THREAD

**Issue by bdovaz** [xoofx/UnityNuGet#331]
@xoofx it has no solution, does it not? I mean, by being MVP and using the free credits they give you, there is no possibility of getting better hardware without it hurting you too much, is there?

Another related question and the main problem that I don't understand why it happens is, in every deployment if I don't change the registry version why does it force to make a deployment from scratch? Shouldn't I have a previous cache when using the same App Service?

Right now it has been going on since 00:00 and is only 31%.

**Comment by xoofx** [comment]
> there is no possibility of getting better hardware without it hurting you too much, is there?

I was on an old service plan, so I tried to upgrade to a better one (77€/month that should be covered by MVP, but I will have to double check that I don't get charge more, as it happened last year - e.g lost 600€+ because of being mischarged)

It restarted.

> Another related question and the main problem that I don't understand why it happens is, in every deployment if I don't change the registry version why does it force to make a deployment from scratch? Shouldn't I have a previous cache when using the same App Service?

I don't know. If it prints `Registry version changed to x.y.z - Regenerating all packages` then that means that the version changed or things on the disk changed or there is a bug in detecting the version number.

UnityNuGet is frankly not viable in the long run. The more we are adding new NuGet packages, the more it is causing a huge burden to replicate all these packages.

At some point, I'm wondering if it should not be self-hosted by folks in company that want to have this, with a trimmed down list of packages only used there.

**Comment by bdovaz** [comment]
@xoofx it's clear that we're all waiting for the amazing work you're doing with the upgrade from Unity to CoreCLR + native NuGet support so we don't have to keep using UnityNuGet but in the meantime it's a lifesaver.

To unblock the problem I'm going to upload the minimum version of the packages (AWSSDK / Google.Apis) that have almost 1000 releases and if someone complains there's no choice but to upgrade those dependencies to a recent one... At least this "patch" will save us for a long time.

**Comment by xoofx** [comment]
> To unblock the problem I'm going to lift the minimum version of the packages (AWSSDK / Google.Apis) that have almost 1000 releases and if someone complains there's no choice but to upgrade those dependencies to a recent one... At least this "patch" will save us for a long time.

Yeah, the AWSSKD/Google.Apis are crazy, so any workaround with them could definitely help.
